docs/developer-guides/merkle-root-construction.md

```diff
@@ -2,10 +2,45 @@
 
 ---
 
-#### Merkle Trees
+## Merkle Trees
 
-The merkle root is constructed using all the TXIDs of transactions in
-this block, but first the TXIDs are placed in order as required by the
+
+A merkle tree is a hash-based data structure that summarizes all the transactions in a block. It can be used to quickly and efficiently verify whether or not a transaction is included in a block. 
+
+Merkle trees are created by repeatedly hashing pairs of nodes until there is only one hash left, the Merkle Root. They are constructed from the bottom up, from hashes of the leaf nodes (hashes of the raw input data). The diagram below shows a merkle root constructed from eight leaf nodes. 
+
+![Merkle Tree](../img/core-blockchain-concepts/merkle_tree.svg)
+
+When calculating the merkle roots included in the block header, the data items are the transaction IDs of the transactions in the block.
+
+Regular transactions and stake-based transactions (ticket purchases, votes, revocations) have separate merkle trees. Regular transactions are hashed to the `MerkleRoot` tree. Stake-based transactions are hashed to the `StakeRoot` tree. 
+
+### Hash Function
+
+Unless otherwise specified, all hashing operations **must** be performed with the BLAKE-256 hash function with 14 rounds.
+
+## Merkle Root Construction
+
+All Merkle trees **must** be constructed according to the unique Merkle tree construct originally implemented by Satoshi, which relies on construction of the tree from an ordered list as follows:
+
+1. Calculate list of BLAKE-256 hashes of original data (e.g. transaction IDs) 
+1. If the list is empty, return a 32-byte root hash of all zeroes; otherwise
```

Comment:
@davecgh - should the second step here be putting the calculated hashes into order? I believe this is critical, otherwise the generated merkle root will not be deterministic.

Reply:
I notice that this copy is taken directly from DCP-0005, so if we make a change here we probably want to make it there also.

Reply:
So most of the equivalent [page](https://bitcoin.org/en/developer-reference#merkle-trees) from the Bitcoin devdocs (which we used as a template) concerns itself with this ordering and its meaning. Wondering if that is overkill, or if we should do the same? Bitcoin content is left at the bottom of the page for reference.

Reply:
The hashes are not sorted.  What the DCP is referring to is that order of the hashes is important to providing their membership which is not typical for generalized set membership testing.

Reply:
The hashes are not sorted.  What the DCP is referring to is that the order of the hashes is important to proving their membership (hence it's referred to as an ordered list, not a sorted list).  It is called out because list order is typically not important for generalized set membership testing and thus it's possible to achieve more efficient encodings in general by discarding the order information.